ocaml/parsing/parser.mly

```diff
@@ -363,23 +363,10 @@ let ptyp_lttuple loc tl =
     ~loc:(make_loc loc) ~attrs:[]
     (Lttyp_tuple tl)
 
-let arg_to_tuple_component (arg_label, body) =
-  let label =
-    match arg_label with
-    | Nolabel -> None
-    | Optional _ ->
-        raise Syntaxerr.(Error(Optional_tuple_component(body.pexp_loc)))
```

Comment:
Looks like `Optional_tuple_component` is now dead code and can be removed.  I considered adding some special cases to give good errors for this, but it doesn't seem worth it.